app/views/hub/notes/_note.html.erb

```diff
@@ -1,4 +1,4 @@
-<li class="note message message--<%= note.user == current_user ? "outgoing" : "incoming"%>">
+<li class="note message message--<%= note.user == current_user ? "outgoing" : "incoming"%>" id="<%= "last-item" if local_assigns[:is_last_item] %>">
```

Comment:
is it necessary to use local_assigns in the partial? can we check based on `is_last_item`'s presence directly. Maybe local_assigns is necessary since is_last_item is an optional parameter and may be undefined?

Reply:
> Maybe local_assigns is necessary since is_last_item is an optional parameter and may be undefined?

I think so. This was my conclusion because it would be undefined unless passed. But honestly I don't like it that much. I just couldn't come up with a better solution. I looked for other options and this was the best I could find in a short amount of time.

Reply:
@bengolder I think the pattern I've used before for optional params is to set a "default value" in the partial using erb, where it would be `<%= some_variable ||= false %>`. I don't know that that's any better, but I also don't know that I knew you could retrieve from local_assigns.